### DISCUSSION TITLE
build questions and minor feedback.

### DISCUSSION THREAD

**Discussion by eMadman** [doudar/SmartSpin2k#247] [General]
I'm about to start working on soldering up the PCB and beginning testing.  Before I break out my soldering iron and start shopping for screws, I have a few questions:

- Looking at the PCB, it looks like it's calling for 12v to the power header while the non-pcb BOM is calling for a 24v supply.  Is it safe to assume that the PCB design is okay to use either 24 or 12v?  I'm knowledgeable enough to know how to make a lot of magic smoke, but looking at the PCB in kicad, it looks like the high voltage is only being delivered to the stepper driver and the recom supply is being used to deliver 5v to the ESP32.  In my experience with 3d printer stuff, the TMC and the motor can definitely handle either 12v or 24v with more torque and quieter operation at 24v.  
- what size screws do I need for the 2.0 case?  Looks like I'll need some pretty long wood screws to get through all of it... For a future iteration, I might modify the 3d model to accept smaller machine screws and use brass heat-set inserts for the threads.  If this project works for me, I'll probably be making at least one or two more for some friends and that mod might be worth my time.
- took me a while to figure out how the design is intended to mount to the bike - the arm for the mount cracked down the layer lines for me trying to get a clean fit.  I Might have to enlarge the part by ~2% to make it more snap-fit friendly for ABS plastic.  
- Same challenge with the stepper motor gear - it meshes beautifully with the larger gear, but doesn't fit the stepper's shaft.  Easy fix with the cad you've provided.  I'll try using a heat gun to soften the plastic before I go with re-printing
- Have you had any concerns with how the stepper rests in the case?  THe lack of screws holding it down worries me since it might introduce some play.  I can't come up with a better mounting solution  besides putting a hole in the chassis and securing the stepper to the case from the rear.  I have some 1/8" thick foam gasket tape that might be a good alternative and give the motor more cushion to sit on.  I think It'll squish as much as necessary for this purpose and do a good job at reducing rattle from the motor not being screwed in place.

I'll post more here through the course of my work on this.  End goal is to use SS2k with QZ and zwift.

**Comment by doudar** [comment]
Sorry for the delay - I don’t get notifications for discussions…

Yes, the pcb can accept up to 24v.

I’ll get back to you on the exact screw sizes for 2.0 (I’m traveling) but they are a common drywall screw size. The reason was just to try to simplify the BOM. 

If you get the exact motor specified, there is no play in the stepper motor. It fits perfectly tight. Very nice actually. 

The original case design has stepper mounting screws but 2.0 is better in this regard. 

Let me know if you make/need any hardware changes- I’d absolutely love a hardware push request if you make some improvements.

Have Fun!

**Reply by eMadman** [reply]
Awesome!  I managed to solder a board together last week, but unfortunately, I can't get the motor spinning using Qdomyos-Zwift and the gear selection there.  I also only just recieved the RCA cables this morning so I can finally wire up the switches.

Best case, it's me misunderstanding how the QZ integration works.  Worst case, I messed up on soldering the tmc or have a bad one (wouldn't be my first defective stepper driver from BTT).  Either way, If that's the case, I'll consider this board a write-off and potentially practice my desoldering with it at a later date.

I'm also having a thought to use female dupont headers to make this type of troubleshooting much easier... not a fan of the huge amount of extra space that would need in the chassis though.

I will definitely provide updated fusion 360 files if I make changes.  The steppers I bought had a JST-like connector which interferes with the motor mount - I was able to trim the mount with my flush cutter - it's just a few mm of material to shave off.  I'll take measurements and make that change for sure.  

I managed to get wood screws that fit the chassis as well (8 x 1.75")

**Comment by eMadman** [comment]
Build complete - wasn't my soldering, just my understanding of how it integrates with QZ or just an issue with QZ on my tablet.  Anyways, it pairs well with Zwift, which is great!

First ride yesterday went okay - about 15-20 minutes in, I noticed that Zwift no longer showed the correct cadence.  Through the logger view on the browser, the SS2k is pulling the correct cadence.  I'll try again tonight using a tablet directly on my spin bike instead of a windows PC.  It's possible there's interference between the frame of the bike and the PC

Since I'm intending on using this with my schwinn IC4 almost exclusively, I'm thinking the better approach to the SS2k is to redesign the chassis to mount to my bike in a more sturdy way.  The stem tube is a square box tube with the corners along the axis of the bike which would make a perfect thing to align the ss2k off of and reduce some slop in it.  

@doudar - my next question:  I noticed that the SS2k paired with my IC4 and automatically started capturing watts which is great... the ic4 calibration will be real difficult for me since I live in an urban center.  Are you at all familiar with the Qdomyos-Zwift integration and what's entailed there?  My hope with this project is to have SS2k pair with QZ and have QZ integrate with Zwift as the power meter since that's what I've used as my baseline for calculating power up to this point.

**Reply by doudar** [reply]
It should work with QZ if you're using the latest version of each. I know they have a few (SS2K) users over there and so maybe you'll want to talk to them to see if there's anything you need special.  

Which version of the SS2K did you build? Version 1 or Version 2?

**Reply by eMadman** [reply]
Thanks!  I reached out on the QZ facebook group to ask questions.  

I built version 2.  Whole process was pretty straightforward.  I've got a couple days off this week so I'm hoping to spend a few hours in fusion 360 and try reprinting it.  I'm terrible at parametric modeling so it'll be slow going, but you've given a great set of files to work with as a starting point.

**Comment by eMadman** [comment]
![image](https://user-images.githubusercontent.com/24726844/142778137-694c5e63-6d00-4d16-932e-53512aece401.png)

In progress: IC4 specific mount.  Length of the crossbar still needs to be measured and figured out.  

Done: Case mod to support my amazon motor and switch to heat set inserts.  https://github.com/eMadman/SmartSpin2k/tree/develop/Hardware/MODS/Case%20V2%20-%20Heat%20Set%20Inserts

I'm still working out issues with integration with Roberto on the QZ side of it, but there's been good progress so far!

**Reply by doudar** [reply]
That's looking awesome! Nice work!

I'm testing a new branch that should have a release shortly. It's much more stable than the last release and reverts to some better ERG code as well. That might help with some issues you're seeing with QZ.

**Reply by doudar** [reply]
Here's an early binary if you'd like to upload and test it (u:admin/p:admin )
[firmware.zip](https://github.com/doudar/SmartSpin2k/files/7577962/firmware.zip)

**Reply by eMadman** [reply]
Thanks! - should this binary still appear as 1.8.22-develop?

**Comment by doudar** [comment]
it should show as 1.8.22-december21Updates

**Reply by eMadman** [reply]
Took a couple of attempts but I got it to flash.  It doesn't seem to want to store any settings changes though:
```
N 227270 HTTP_Server: "Link Not Found: /connecttest.txt"
N 235916 Main: "Stepper power is now 1500"
N 235930 Main: "Stealthchop is now 0"
N 235936 HTTP_Server: "Config Updated From Web"
N 235946 Config: "Writing File: /config.txt"
E 236057 Config: "Failed to create file"
N 236057 Config: "Contents of file: /config.txt"
N 236082 Config: "Writing File: /userPWC.txt"
E 236182 Config: "Failed to create file"
N 236182 Config: "Contents of file: /userPWC.txt"
```

**Reply by doudar** [reply]
Most likely a file got uploaded that wasn't named "firmware.bin" and it is taking up all the space in your spiffs partition. The easiest is to reset to defaults, reconnect to the network (so it can download the html files) and then flash again to the test binary via the web interface.

**Reply by doudar** [reply]
...and here's another new binary. Lots of stability improvement and I fixed a memory leak in one of the newer routines. It should be very stable now. Remember to unzip it before you test. 
[firmware.zip](https://github.com/doudar/SmartSpin2k/files/7592656/firmware.zip)

**Reply by eMadman** [reply]
Nice!  I'll do some testing tonight.  I owe Roberto over at QZ some logs as well, so I'm excited to see how it goes.  

RE: the write issues I was seeing, I eventually had to erase the flash on the esp32, but it appears to be taking firmware updates now

**Reply by doudar** [reply]
Version 1.11.24 just went live on the automatic updates. Please send any feedback you have.

**Comment by eMadman** [comment]
I couldn't get QZ to update/manipulate resistance on SS2k in my ride last night with Zwift which is the same issue I've been working on.  

I did have success pairing SS2k to my IC4 and getting output that way.  Compared to when I started this about 2 weeks ago, I think the experience was a solid success.  Readings were stable and I didn't experience any latency.  Last attempt had zwift fall out of sync not too far into the ride.  

I think the readings are way wrong and ended up setting the correction factor around 0.6 to compensate.  Else it had me doing 35-40kph on flat roads... I'm pretty confident I'm not nearly that fast.  I know that's not a proper calibration but it worked great as a proof of concept.  I'm very likely going to go down the path of buying and mounting a power meter in order to pull software-calculated wattage out of the equation.  Directly connected to zwift, this worked pretty nicely.  It was neat to watch resistance dancing up and down through the ride.  Latency didn't appear to be a significant issue either.

**Comment by doudar** [comment]
Great to hear things are working for you! 

Maybe I'll have to grab QZ someday and throw it on an old phone for testing/debugging.

**Comment by eMadman** [comment]
Well, I'm definitely buying a power meter now.  This schwinn/Bowflex mount works brilliantly and the wobble is significantly reduced.  [Quick video of mount and dismount](https://photos.app.goo.gl/55tayBfx3VSynJ28A)

Using some very thick command strips right now to hold it up and I know they're adding a lot of flex to the system.  Next version will be better suited for zip ties, but will need some rubber/silicon anti-skid dots to stop it from sliding around.  This version could be attached more permanently and rigidly with VHB tape, but I was worried it would damage the paint.

Also need to add holes to be able to see the esp32's LED's.  For now, you can see them if you peek into the USB port.

![image](https://user-images.githubusercontent.com/24726844/144154082-946314dc-77e9-4369-8a4d-50c85c6263d2.png)

Friction fit - tight tolerance on the print - currently using 1mm shim washers to add tightness.  Next version will close the gap and get rid of the washers.  That should make it a bit more rigid and last longer.  The superglue holding the washers in place probably won't last long.
![image](https://user-images.githubusercontent.com/24726844/144154111-70023323-28be-45eb-903a-97bb79305001.png)

**Reply by doudar** [reply]
Wow! Great pictures - those look awesome and the mount is very nice!

**Reply by doudar** [reply]
> Well, I'm definitely buying a power meter now. This schwinn/Bowflex mount works brilliantly and the wobble is significantly reduced. [Quick video of mount and dismount](https://photos.app.goo.gl/55tayBfx3VSynJ28A)
> 
> Using some very thick command strips right now to hold it up and I know they're adding a lot of flex to the system. Next version will be better suited for zip ties, but will need some rubber/silicon anti-skid dots to stop it from sliding around. This version could be attached more permanently and rigidly with VHB tape, but I was worried it would damage the paint.
> 
> Also need to add holes to be able to see the esp32's LED's. For now, you can see them if you peek into the USB port.
> 
> ![image](https://user-images.githubusercontent.com/24726844/144154082-946314dc-77e9-4369-8a4d-50c85c6263d2.png)
> 
> Friction fit - tight tolerance on the print - currently using 1mm shim washers to add tightness. Next version will close the gap and get rid of the washers. That should make it a bit more rigid and last longer. The superglue holding the washers in place probably won't last long. ![image](https://user-images.githubusercontent.com/24726844/144154111-70023323-28be-45eb-903a-97bb79305001.png)

BTW, are you able to push the mount and maybe some instructions to the mods folder? The IC4 is a popular bike and this mount looks pretty hot.

**Reply by eMadman** [reply]
I'm a git newbie and am maintaining it in /hardware/mods on [my fork here](https://github.com/eMadman/SmartSpin2k/tree/develop/Hardware/MODS/Case%20V2%20-%20Heat%20Set%20Inserts).  Need to push my latest STL's and CAD.  ~~I've got one final revision to make and it'll be as done as I plan to get it until I get the power meter - will probably do that over the next few days and push the changes again.~~

I think this design should be easily adaptable for other bikes - just have to change the dimensions of the mount and potentially print a bigger arm.

I hope the heat set inserts aren't a bit of a stretch for users.  Love these things.  Either way

@doudar - I figured out how to create the PR.  Messed up a bit trying to revert back to the original readme.md from the main repo, but I think I got it right

**Reply by fireyeti** [reply]
Great mount and awesome work! I really like the slide on the crossbar for adjustable length on various models.

**Reply by doudar** [reply]
> I'm a git newbie and am maintaining it in /hardware/mods on [my fork here](https://github.com/eMadman/SmartSpin2k/tree/develop/Hardware/MODS/Case%20V2%20-%20Heat%20Set%20Inserts). Need to push my latest STL's and CAD. ~I've got one final revision to make and it'll be as done as I plan to get it until I get the power meter - will probably do that over the next few days and push the changes again.~
> 
> I think this design should be easily adaptable for other bikes - just have to change the dimensions of the mount and potentially print a bigger arm.
> 
> I hope the heat set inserts aren't a bit of a stretch for users. Love these things. Either way
> 
> @doudar - I figured out how to create the PR. Messed up a bit trying to revert back to the original readme.md from the main repo, but I think I got it right

All good- you did great- thx!!!

Now you need to start building these for all those guys over on the QZ Facebook page 😅

**Reply by doudar** [reply]
Also, I don't know what program you're using to CAD with but I can easily send a different master format to you if you ever want to do any other modding.

**Reply by eMadman** [reply]
I'm on fusion 360.  The CAD you provided for the case and gears was perfect for my needs as a start point.  I would have been screwed if you used any organic shapes or asymmetrical geometries.  You should see the timeline history on my changes to the file - I genuinely don't know how I managed to keep things organized in my head!

In terms of further modding, I think the case side is more or less where I like it.  The mount point for the arms could be moved up onto the curved surface to make room for the LED's to shine through but I chose not to mess with that since I wasn't sure how strong it'd be up there.   I also didn't want to interrupt the one large, round feature on the design since I think it looks nice and breaks up the blockiness.

The bike mount for the schwinn/bowflex is pretty close to perfect... a bit more corner and sharp edge rounding never hurts.  I got the final revision mounted up last night with zip ties to replace the command strips and it appears to be rock solid.  The square corner where it fixes against the bike should be easy to replace with other geometries to suit other bikes - longer wings for the corner style, half moon for round tubes.

The sliding arms can be rounded out a touch more at the case side and moved further away from the case to allow more articulation for different bikes.  Of course, a user could simply turn the orientation Might also be worth making the slide thicker (requiring changes to the bike mount and case) to make it stiffer in case longer lengths are needed.  At 7.3mm overall and 3mm at it's thinnest, it will probably be a bit too bendy for PLA and potentially ABS at longer lengths.

Here's a visual on where I'm thinking of going:
![image](https://user-images.githubusercontent.com/24726844/144437113-13749c95-1bcf-41af-9d37-742341361be4.png)


Pipe dream idea:  I had the wild idea of having it belt-driven with some spare gt2 pulleys I have sitting around, something similar to the drive gear setup on a Voron Mobius extruder or the Z drives for the V2 machines, but I genuinely doubt that change would make anything "better" for this use case - just a cool concept.  We're not pushing enough torque to need anything that ridiculous.... voron z drives have to handle a massive and very heavy flying gantry without sagging too much when you kill power to the steppers.

**Reply by doudar** [reply]
I've thought about belt drive as well but I abandon it in the effort to keep the (mail order) parts count as low as possible. Adding a belt would add two pulleys, a gates belt, and changing the output shaft to a metal one. In addition to the lower parts count, why I used a printed shaft on the main gear instead of a 5/16" bolt was, in addition to keeping the parts count low, it also provides a mechanical fuse making a low possibility of damaging a bike (by turning too far/hard) or injuring a child.  

Surprisingly that little plastic shaft handles the torque well and is robust in that effort, but I've recently had an instance where I was installing the knob cup onto a new insert that was a little too tight and put a side load on the shaft and broke it. Case 2.0 makes changing that gear super simple but maybe it's time to revisit adding the 5/16" bolt shaft. 

The current gearing provides a good amount of torque (enough to move my felt resistance test bike full scale) but is nowhere near the (expensive) prototype 24:1 planetary gear drive so harming or breaking someone/something is still improbable . That means my  reasoning behind a "mechanical fuse" is out of date and no longer needed. Case 2.0 also greatly reduced the parts count so I think the benefit of increasing durability by adding an (easy to source) 5/16" bolt outweighs the increased parts count.  

To modify the SS2K to a metal shaft, the output gear and knob cup would both just need hex recesses and a 5/16" bolt of the proper length to act as a shaft. It would also require bit of thought as to nut retention (maybe a double nut would suffice?) It could probably be done in a shorter amount of time than it took me to write this post. 

My pipe dream is getting the PCB to a point where anyone can just order a completed board from a pcb manufacturer. I'm nearly to the point of ordering a prototype but I've been to this point before this year and keep getting derailed by shortages which cascade design changes (and I'm new to PCB design.)

The new board is going to have a TMC2209 on it which will enable development sensor less homing and stall detection - something that's been on the list for a while and might enable some nice features.

**Reply by eMadman** [reply]
Looks like you gave belt drive a lot more thought than I did.  Designing for a belt drive is far outside my skill set.  It's a cool engineering achievement, but yea... I don't think it's necessary for the application.  Smaller packaging while maintaining the current torque though... that could be cool!  There's some potential rabbit holes with high powered pancake steppers from LDO - the ones popularized by the Orbiter/Galileo/Sherpa mini extruders are available with [standard shafts and are absolutely teeny tiny](https://www.printedsolid.com/products/ldo-nema14-36mm-round-pancake-motor-ldo-36sth17-1004ah).  Much harder to source than a standard nema 17 though. I have one in my parts bin... never got around to making the extruder it was intended for.

[take a look at these](https://www.aliexpress.com/item/4000303705414.html?spm=a2g0o.productlist.0.0.53ee6306Hqo5tb&algo_pvid=b90c9469-acd9-4202-8017-3e2c971a9954&algo_exp_id=b90c9469-acd9-4202-8017-3e2c971a9954-0&pdp_ext_f=%7B%22sku_id%22%3A%2210000001254836800%22%7D) for metal shafts with pre-ground flats to handle grub screws.  I've got 'em all over my Voron.  For the application, they might be a bit pricey - I seem to remember them being much cheaper once upon a time.

The TMC 2209 however - that would be be amazing and would end the grind of doom that happened occasionally when I had the buggy firmware earlier last month.

FYI, I posted pictures on the IC4 facebook group and it gathered a ton of interest.  If it's okay, I'd like to make and sell a few units with my spare/extra parts to get more feedback from the users.  Looking at the license, I think that would be okay, but I'd like to be transparent on it as well.

**Reply by doudar** [reply]
By all means, build and sell to your hearts content! My only stipulation is that  if I'm ever in town you buy me a beer :)

Edit: Stipulation 2 :)  and try to drive more interest into the project so we can get more talent and development.

**Reply by doudar** [reply]
Also, @eMadman I'm going to ping @marklogan17 and @cagnulein this discussion as I know they had a lot of interest over there on the QZ page looking for builders as well. (and I'm just too busy to be building any currently)

**Reply by rjbutler** [reply]
Hey @eMadman if you have the STL files for the IC4 case/arm, I'd love a copy.  Reproducing it is beyond my current abilities.

I did get my SS2K up and running on my second attempt using a TMC 2208 instead of the TMC 2225 since those are hard to come by.  I'm pretty sure the TMC 2225 I had was bad and what cause the first attempt to fail.

Thanks

**Reply by eMadman** [reply]
@rjbutler - You can get the STLs from the [MODS folder here](https://github.com/doudar/SmartSpin2k/tree/develop/Hardware/MODS/Case%20V2%20-%20Heat%20Set%20Inserts).  I apologize in advance - I used threaded heatset inserts for the design because I love how repeatable they are.  Links are in the readme.md - totally worth having a sack of a hundred or so because these things are incredibly useful and give a stronger screw-together bond than most other fasteners for printed plastic.  

I need to rename that folder to IC4 or something and update my bom to include the following:

Will also need:
- 4xM3 screws - I think M3x10 should work, possibly m3x12?  As long as it's long enough to go into the threaded inserts, the length doesn't matter.  
- 1 pc M5x10mm for the slide on the arm
- 2pc M5x30mm - 1 for the bike mount and another for the case
- 3pc M5 nuts
- 2pc M5x1mm shim washers for the case - if you don't have accurately thick 1mm washers, this can be printed by making a 1mm tall cylinder with a 5mm hole.  Shim washers tend to be precision sized to the advertised thickness - much nicer than typical stamped ones. My next update of the case will eliminate these from the BOM because I don't believe it's needed.


For the bike mount, you have two options:  VHB tape or something similar OR if you want to protect the paint, you can use the zip tie and rubber sticker approach:

- 6x zip ties for the bike mount - what I did here is fish a zip tie from one end, around the square tube and through the other.  Used a 2nd zip tie's ratchet bit to tighten the first zip tie.  Trim off the excess.  Between the rubber and zip ties, it's a pretty solid mount
- some sort of rubber or silicon non-skid stickers to put on the bike mount to add friction

@doudar - Come on up to Toronto and I'll definitely crack one open with you!  I've also shared the idea to a couple of friends and have a buddy lined up to help with testing further - dude's an absolute data nerd and a perfectionist so he'll be all over it from the standpoint of getting it perfect.

**Reply by cagnulein** [reply]
@eMadman you should consider to post something on the Schwann Facebook group about this! They already know QZ so the step will be close!

**Reply by rjbutler** [reply]
Thanks @eMadman!  I'll see about printing it this weekend.   What I have works, but it moves side to side a lot more than I'd like

**Reply by eMadman** [reply]
@cagnulein - already done!  All my spare parts are spoken for by people expressing interest in it.  This is great because I haven't had the time I needed to get you updated logs on the bug ticket I have open on the SS2k repo - might get a couple more people helping test the software integration of it this way.  I'm hoping to start getting them mailed out before Christmas

**Reply by doudar** [reply]
> @cagnulein - already done! All my spare parts are spoken for by people expressing interest in it. This is great because I haven't had the time I needed to get you updated logs on the bug ticket I have open on the SS2k repo - might get a couple more people helping test the software integration of it this way. I'm hoping to start getting them mailed out before Christmas

Shoot me an email to smartspin2k@gmail.com if you need help sourcing parts. I’ve got a decent stockpile and I’m not planning to build any for a while (working on printer mods :) )
